sky/cli.py

```diff
@@ -2108,10 +2108,15 @@ def cancel(cluster: str, all: bool, jobs: List[int], yes: bool):  # pylint: disa
             arg_str = '--all'
         else:
             arg_str = ' '.join(map(str, jobs))
-        error_str = ('Cancelling the spot controller\'s jobs is not allowed.'
-                     f'\nTo cancel spot jobs, use: sky spot cancel <spot '
-                     f'job IDs> [--all]'
-                     f'\nDo you mean: {bold}sky spot cancel {arg_str}{reset}')
+        if cluster == spot_lib.SPOT_CONTROLLER_NAME:
+            error_str = (
+                'Cancelling the spot controller\'s jobs is not allowed.'
+                f'\nTo cancel spot jobs, use: sky spot cancel <spot '
+                f'job IDs> [--all]'
+                f'\nDo you mean: {bold}sky spot cancel {arg_str}{reset}')
+        else:
```

Comment:
should we do `elif` to make sure it actually matches the serve controller name?

Reply:
Does an `assert` sound good to you..? The error should only be trigger if it is a spot controller or a skyserve controller